UDT.Business/Implementation/SubjectService.cs

```diff
@@ -88,5 +89,16 @@ public async Task<Subject> UpdateAsync(Subject updatedSubject)
 
             return existingSubject;
         }
+
+        public async Task<User> GetTeacherForSubject(int id)
+        {
+            var existingSubject = await _dataContext.Subjects
+                .Include(subject => subject.Users)
+                .FirstOrDefaultAsync(subject => subject.Id == id);
+
+            if (existingSubject is null) return null;
+
+            return existingSubject.Users.Find(user => user.Role == UserRole.Teacher);
```

Comment:
This will throw an exception if it doesn't find any user matching the Predicate. Because we don't manage the exceptions, you could use FirstOrDefault()